### PR TITLE
expose and document run_data and scan_data on nodes

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description
https://github.com/chef/automate/issues/192

**this work should only merge AFTER https://github.com/chef/automate/pull/332 and https://github.com/chef/automate/pull/345**

this is to expose the new fields in the gateway and document the new filtering options in the nodes api

### :+1: Definition of Done
nodes api new filtering options are documented; data exposed

### :athletic_shoe: Demo Script / Repro Steps
rebuild gateway
send some nodes in from chef_load_nodes
send some nodes in from chef_load_compliance_scans
try some of the api calls

### CODE REVIEW

File: components/automate-chef-io/content/docs/nodes.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ When a user creates a node, that node is added to the `/nodes` endpoint.
 
 When a user adds a node integration, like aws or azure, nodes are added to the `/nodes` endpoint.
 
-When an inspec report is ingested, a node is added to the `/nodes` endpoint. If it already exists, the last contact time is updated.
+When an inspec report or chef client run is ingested, a node is added to the `/nodes` endpoint. If it already exists, the last contact time is updated, along with run data and scan data, that stores the latest run id or report id and status (as well as the penultimate status).
```
Comment (phiggins) [inline components/automate-chef-io/content/docs/nodes.md]:
```suggestion
When an inspec report or chef client run is ingested, a node is added to the `/nodes` endpoint. If the node already exists, its last contact time, run data, and scan data are updated, storing the status and the latest run id or report id, and the penultimate status.
```

That last sentence has a lot going on. This is my attempt to clean it up. WDYT?

File: components/automate-chef-io/content/docs/nodes.md [hunk 2]
```diff
@@ -57,9 +57,72 @@ The `/nodes` endpoint supports filtering by:
 - region
 - source_id (a reference to the primary provider's node)
 - state
-- statechange_timestamp
+- statechange_timerange (supports two timestamps of type "2019-03-05T00:00:00Z")
```
Comment (phiggins) [inline components/automate-chef-io/content/docs/nodes.md]: Does the timestamp conform to any standard, eg ISO8601?

File: components/automate-chef-io/content/docs/nodes.md (hunk 1 above)
Comment (mjingle) [inline components/automate-chef-io/content/docs/nodes.md]: How is this? 
```suggestion
When an InSpec report or a Chef Infra Client run is ingested, a node is added to the `/nodes` endpoint. If the node already exists, its last contact time, run data, and scan data are updated. When the run data and scan data are updated, the latest information is stored for the run id or report id, the status, and the penultimate status.
```

File: components/automate-chef-io/content/docs/nodes.md (hunk 2 above)
Comment (vjeffrey) [reply components/automate-chef-io/content/docs/nodes.md]: urr..good question.

Comment (lancewf) [reply components/automate-chef-io/content/docs/nodes.md]: RFC3339 is the date time standard that Chef tries to use.

File: components/automate-chef-io/content/docs/nodes.md [hunk 3]
```diff
@@ -57,9 +57,72 @@ The `/nodes` endpoint supports filtering by:
...
 - status
 - tags
+- last_run_timerange (last time reported on ingested ccr: supports two timestamps of type "2019-03-05T00:00:00Z")
+- last_scan_timerange (last time reported on ingested scan: supports two timestamps of type "2019-03-05T00:00:00Z")
+- last_run_status (status on last ingested ccr)
+- last_scan_status (status on last ingested scan)
+- last_run_penultimate_status
+- last_scan_penultimate_status
```
Comment (lancewf) [inline components/automate-chef-io/content/docs/nodes.md]: Can we add a comment here for the `penultimate` fields? I had to look up what 'penultimate' was to figure out what these provide. 
`last_run_penultimate_status (status on second to last ingested ccr)`
`last_scan_penultimate_status (status on second to last ingested scan)`

**Pull request merged**